src/physics/initial_conditions/initial_condition_function.cpp

```diff
@@ -14,6 +14,43 @@ ::InitialConditionFunction ()
     // Nothing to do here yet
 }
 
+template <int dim, int nstate, typename real>
+real InitialConditionFunction<dim,nstate,real>
+::convert_primitive_to_conversative_euler(const std::array<real,nstate> primitive_value,
```

Comment:
I understand that TGV kind of had the `convert_primitive_to_conversative_value` and you mimicked that part. We are effectively copy pasting some Euler code here and in Physics::Euler. Could you instead make it so that the derived classes that do use Euler equations simply hold a Physics::Euler (or NavierStokes if it makes more sense) and call the primitive to conservative function?

This effectively allows us to easily change the physics if needed. For example, what if we decide that Euler can use various gas laws.

Reply:
Also, I just noticed you trying to simplify it by moving it to the base class ;). The thought was there! Minor execution change.

Reply:
Changed by creating a `shared_pointer` to Euler physics in the relevant classes. It's possible that this will be a little less efficient than the previous implementation, but it's cleaner in my opinion. Plus initialization is a small cost compared to solving!

I've left `exact_solution` as-is because the intention is to eventually merge these two classes, and there's no repeated code as only isentropic vortex has an exact solution.